app/models/user.rb

```diff
@@ -64,12 +72,24 @@ def self.from_omniauth(token:)
     return nil if token.blank?
 
     auth = HydraPublicApiClient.fetch_oauth_user(token:)
-    return nil unless auth
+    return nil if auth.blank?
 
     auth = auth.stringify_keys
     args = auth.slice(*ATTRIBUTES)
-    args['id'] = auth['sub']
+
+    args['id'] ||= auth['sub']
+    args['token'] = token
+
+    # TODO: remove once the HydraPublicApi returns the 'organisations' key.
+    temporarily_add_organisations_until_the_profile_app_is_updated(args)
 
     new(args)
   end
+
+  def self.temporarily_add_organisations_until_the_profile_app_is_updated(hash)
+    return hash if hash.key?('organisations')
+
+    # Use the same organisation ID as the one from users.json for now.
+    hash.merge('organisations' => { '12345678-1234-1234-1234-123456789abc' => hash['roles'] })
+  end
```

Comment:
📓 remove when profile application is fully expanded to include the organisations table